dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/RotatingDnsResolver.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.net
+
+import java.net.InetAddress
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.nanoseconds
+import okhttp3.Dns
+
+internal class RotatingDnsResolver(
+    private val delegate: Dns = Dns.SYSTEM,
+    private val ttl: Duration = TTL_30_MIN
+) : Dns {
+
+    data class ResolvedHost(
+        val hostname: String,
+        val addresses: MutableList<InetAddress>
+    ) {
+        private val resolutionTimestamp: Long = System.nanoTime()
+
+        fun getAge(): Duration {
+            return (System.nanoTime() - resolutionTimestamp).nanoseconds
+        }
+
+        fun rotate() {
+            val first = addresses.removeFirstOrNull()
+            if (first != null) {
+                addresses.add(first)
+            }
+        }
+    }
+
+    private val knownHosts = mutableMapOf<String, ResolvedHost>()
+
+    // region Dns
+
+    override fun lookup(hostname: String): MutableList<InetAddress> {
+        val knownHost = knownHosts[hostname]
+
+        return if (knownHost != null && isValid(knownHost)) {
+            knownHost.rotate()
+            knownHost.addresses.toMutableList()
```

Comment:
`ResolvedHost#addresses` already has `MutableList<InetAddress>`, so I think `toMutableList()` call is not needed here

```suggestion
            knownHost.addresses
```

Reply:
This is done in order to not have the list we keep mutated from outside this class